src/helpers/losses.jl

```diff
@@ -609,6 +609,34 @@ function __unsafe_apply_loss(loss::GenericLossFunction, ŷ, y)
     return __fused_agg(loss.agg, loss.loss_fn, ŷ, y)
 end
 
+@concrete struct SAMLoss <: AbstractLossFunction
+    loss_fn
+    ρ
+end
+
+function SAMLoss(loss_fn; ρ=0.05)
+    return SAMLoss(loss_fn, ρ)
+end
+
+
+function (loss::SAMLoss)(model, ps, st, (x, y))
+    ŷ, st_ = model(x, ps, st)
+    return loss.loss_fn(ŷ, y)
+end
+
+function __unsafe_apply_loss(loss::SAMLoss, ŷ, y)
+    return loss.loss_fn(ŷ, y)
+end
+
+function CRC.rrule(cfg::RuleConfig{>:HasReverseMode}, SL::typeof(SAMLoss), model, ps, st, (x, y))
+    grad = Zygote.gradient(pars->SL.loss_fn(model(x, pars, st)[1], y), ps)
```

Comment:
Lux doesn't have a dep on `Zygote` (and can't take one). You can do this by calling `CRC.rrule_via_ad(cfg, ....)`